apigw-lambda-comprehend-cdk-python/src/lambda_function.py

```diff
@@ -0,0 +1,37 @@
+import boto3
+import os
+import logging
+import json
+from botocore.exceptions import ClientError
+
+LOG = logging.getLogger()
+LOG.setLevel(logging.INFO)
+
+region_name = os.getenv("region", "us-east-1")
+comprehend_obj = boto3.client("comprehend", region_name=region_name)
+
+
+def lambda_handler(event, context):
+    """
+    :param event: Input from the user, through API Gateway
+    :param context: Any methods and properties that provide information about the invocation, function, and execution environment
+    :return: THe response from Comprehend service about the Sentiment of the user input.
+    """
+    try:
+        LOG.info(f"Event is {event}")
+        LOG.info(event["body"])
+        user_input = json.loads(event["body"])["input"]
+        response = comprehend_obj.detect_sentiment(Text=user_input, LanguageCode="en")
+        LOG.info(response)
+        LOG.info(f"Sentiment detect is {response['Sentiment']}!!")
+    except ClientError:
+        LOG.error("Sentinment detection failed!!")
+        raise
+    else:
+        analyzed_response = json.dumps(
+            {
+                "Sentiment": response["Sentiment"],
+                "Confidence Score": response["SentimentScore"],
+            }
+        )
+        return analyzed_response
```

Comment:
Question: is there a reason why this code is not in the `try` block?

Reply:
Hm, not sure I follow.

Can you explain why the equivalent w/out elese doesn't this work here? Also, any other exception thrown in your code is not handled (e.g., dict key errors). Is that intentional? If not, consider making exception handling more general (catch any exception, log, and just raise)

```python
    try:
        LOG.info(f"Event is {event}")
        LOG.info(event["body"])
        user_input = json.loads(event["body"])["input"]
        response = comprehend_obj.detect_sentiment(Text=user_input, LanguageCode="en")
        LOG.info(response)
        LOG.info(f"Sentiment detect is {response['Sentiment']}!!")
        analyzed_response = json.dumps(
            {
                "Sentiment": response["Sentiment"],
                "Confidence Score": response["SentimentScore"],
            }
        )
        return analyzed_response
    except ClientError:
        LOG.error("Sentinment detection failed!!")
        raise
```

Reply:
@embano1 - Your assumption is right. It was not intentional. The idea was to execute return the response through else block. 

I've modified the code to catch any exception in the original PR# https://github.com/aws-samples/serverless-patterns/pull/1927